core/src/main/java/com/graphhopper/routing/util/AbstractWeighting.java

```diff
@@ -32,6 +38,40 @@ public AbstractWeighting( FlagEncoder encoder )
             throw new IllegalStateException("Not a valid name for a Weighting: " + getName());
     }
 
+
+    @Override
+    public double calcWeight( EdgeIteratorState edgeState, boolean reverse, int prevOrNextEdgeId )
+    {
+        double weight = calcPureWeight(edgeState, reverse,prevOrNextEdgeId);
+
+        if(adjustments != null){
```

Comment:
btw: lazy initialization here is not good as although calcPath is single threaded in core GH, this can change and others can have multi-threaded algos. A weighting should be seen as read-only.